### PR TITLE
Add bfloat16 softmax and gelu

### PR DESCRIPTION
**Pull request by wozna**
### PR types

New features
### PR changes

OPs 
### Describe

This PR adds bfloat16 support for `softmax` and `gelu` activations function needed for BERT model.

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/mkldnn/test_softmax_bf16_mkldnn_op.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    def init_kernel_type(self):
+        self.use_mkldnn = True
+
+
+class TestSoftmaxMKLDNNOp6(TestSoftmaxOp6):
+    def init_kernel_type(self):
+        self.use_mkldnn = True
+
+
+if __name__ == '__main__':
```
Comment (jczaja) [inline python/paddle/fluid/tests/unittests/mkldnn/test_softmax_bf16_mkldnn_op.py:90]: consider adding enable_static() here if needed

File: paddle/fluid/operators/softmax_op.cc [hunk 2]
```diff
@@ -115,6 +115,11 @@ class SoftmaxOpMaker : public framework::OpProtoAndCheckerMaker {
     AddAttr<bool>("use_mkldnn",
                   "(bool, default false) Only used in mkldnn kernel")
         .SetDefault(false);
+    AddAttr<std::string>(
+        "mkldnn_data_type",
+        "(string, default \"float32\"). Data type of mkldnn kernel")
+        .SetDefault("float32")
+        .InEnum({"float32", "int8", "bfloat16"});
```
Comment (jczaja) [inline paddle/fluid/operators/softmax_op.cc]: I'm not sure softmax int8 will ever exist. Won't there be any risk here that softmax may be tried to be initialzied as int8?

File: paddle/fluid/inference/analysis/ir_pass_manager.cc [hunk 3]
```diff
@@ -79,6 +79,10 @@ void IRPassManager::CreatePasses(Argument *argument,
     } else if (pass_name == "cpu_quantize_pass") {
       pass->Set("quant_var_scales",
                 new VarQuantScale(argument->quant_var_scales()));
+    } else if (pass_name == "cpu_bfloat16_placement_pass") {
```
Comment (jczaja) [inline paddle/fluid/inference/analysis/ir_pass_manager.cc:82]: Why this change is needed?

File: paddle/fluid/operators/softmax_op.cc (hunk 2 above)
Comment (wozna) [reply paddle/fluid/operators/softmax_op.cc]: Yes you right, I will change it.

File: paddle/fluid/inference/analysis/ir_pass_manager.cc (hunk 3 above)
Comment (wozna) [reply paddle/fluid/inference/analysis/ir_pass_manager.cc:82]: It should be added earlier because without it function in C-API `SetBfloat16Op` doesn't work.

Comment (jczaja) [reply paddle/fluid/inference/analysis/ir_pass_manager.cc:82]: ok

**Review by jczaja — APPROVED** [review]
LGTM

**Comment by jczaja** [comment]
@luotao1 could you please start your review?

**Review by luotao1 — APPROVED** [review]
